### PR TITLE
Add s3-multipart-download (resolves #103)

### LINKED ISSUE #103: Containerize: s3-multipart
Needed for downloading large files from Ceph via Boto.

### PR DESCRIPTION
**Pull request by jvivian**
Requires boto credentials in mounted /data dir

Test command
`docker run -v /location/to/boto:/data quay.io/uscs_cgl/sm-multipart-download s3://cgl-driver-projects/test/test.txt /data`

### CODE REVIEW

File: s3-multipart-download/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# File Author / Maintainer
+MAINTAINER John Vivian <jtvivian@gmail.com>
+
+# Update the repository sources list
+RUN apt-get update && apt-get install --yes \
+        git \
+        python-pip
+
+RUN git clone https://github.com/jvivian/s3-multipart /opt/s3-multipart
```
Comment (hannes-ucsc) [inline s3-multipart-download/Dockerfile:12]: Checkout specific commit or tag.

**Comment by hannes-ucsc** [comment]
1) Please move or fork your s3-multipart repo to the BD2KOrganization, rename it s3-multipart-download or s3-mp-download. Then use that name consistently across the board, for image name, script name, install directory etc.

2) And as mentioned above, please refer to a specific commit in the checkout.

**Comment by hannes-ucsc** [comment]
The boto credential requirement should be added to a container-specific README, along with basic usage information.

**Comment by hannes-ucsc** [comment]
Disregard 1).

Can we add MP downloads as a common function to toil-scripts? A container for a single, small Python script seems overkill. If not, then clone to the upstream repo in the Dockerfile (not your fork) and inline your OrdinaryCallingFormat change as a patch file here in this repo. File an issue/PR against the upstream repo and refer to that issue in a comment in the Dockerfile, e.g.

```
# Fix for http://github.com/..../issues/#123
RUN patch -p0 < issue-123-ordinary-calling-format.patch
```

**Comment by jvivian** [comment]
> Can we add MP downloads as a common function to toil-scripts?

I think this is the best approach, although we'll want to modify it so that we can use the same function for both S3 and Ceph (only change `connect_s3()` gets passed `host='ceph-gw-01', is_secure=False, calling_format=boto.s3.connection.OrdinaryCallingFormat(),)`

**Comment by jvivian** [comment]
Not needed — multipart uploading to Ceph will soon be handled by S3AM as well: https://github.com/BD2KGenomics/s3am/issues/45

**Pull request closed without merging**